src/components/Login/Login.jsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react';
+import './Login.css';
+import Button from '../../common/Button/Button';
+import Input from '../../common/Input/Input';
+import { Link, useNavigate } from 'react-router-dom';
+import { LOGIN_LABEL, SERVER_URL } from '../../constants';
+import PropTypes from 'prop-types';
+
+const Login = () => {
+	const [formData, setFormData] = useState({
+		email: '',
+		password: '',
+	});
+
+	const [formErrors, setFormErrors] = useState({
+		email: '',
+		password: '',
+		server: '',
+		login: '',
+	});
+
+	const handleInputChange = (event) => {
+		const { name, value } = event.target;
+		setFormData({
+			...formData,
+			[name]: value,
+		});
+	};
+
+	const navigate = useNavigate();
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		const requiredErrors = {
+			email: formData.email ? '' : 'Email is required',
+			password: formData.password ? '' : 'Password is required',
+		};
+		if (requiredErrors.email || requiredErrors.password) {
+			setFormErrors(requiredErrors);
+		} else {
+			const response = await fetch(`${SERVER_URL}/login`, {
+				method: 'POST',
+				body: JSON.stringify(formData),
+				headers: {
+					'Content-Type': 'application/json',
+				},
+			});
+
+			const result = await response.json();
+			if (result.successful) {
+				localStorage.setItem('userToken', result.result);
+				localStorage.setItem('userName', result.user.name);
+				if (result.user.name.toLowerCase().includes('admin')) {
+					localStorage.setItem('isAdmin', true);
+				}
+				navigate('/courses');
+			} else {
+				setFormErrors({ server: result.result, login: result.errors });
+			}
+		}
+	};
+
+	// FIXME - is there any other way to write large forms ?
```

Comment:
Usually creates component like FormItem whereby the type(input, checkbox, select) be rendered some form controller